pytest_socket.py

```diff
@@ -40,14 +40,13 @@ def pytest_addoption(parser):
 
 @pytest.fixture(autouse=True)
 def _socket_marker(request):
+    if request.config.getoption('--disable-socket'):
+        request.getfixturevalue('socket_disabled')
     if request.node.get_closest_marker('disable_socket'):
         request.getfixturevalue('socket_disabled')
     if request.node.get_closest_marker('enable_socket'):
         request.getfixturevalue('socket_enabled')
 
-    if request.config.getoption('--disable-socket'):
-        request.getfixturevalue('socket_disabled')
-
```

Comment:
any comments how this fixes the issue? (if there is an ordering dependency we should probably add comments explaining it)

Reply:
It's an ordering issue. The things higher on this list have lower priority. The last thing to set the fixture value wins.

Generally for the purposes in which someone would be using this package (IMHO), people expect more granular settings to have highest priority.
I.E. a marker on a particular function overrides the global disable flag.

This is also the documented and expected behavior; the readme states:
>To enable specific tests use of socket, pass in the fixture to the test or use a marker

I can add a bit of comments to the function here.